src/scan_shadow_detector.cpp

```diff
@@ -42,10 +42,21 @@
 
 namespace laser_filters
 {
-  void ScanShadowDetector::configure(const float min_angle, const float max_angle)
+  void ScanShadowDetector::configure(const float min_angle, const float max_angle, const int window)
   {
     min_angle_tan_ = tanf(min_angle);
     max_angle_tan_ = tanf(max_angle);
+    window_ = window;
+    angle_increment_ = 0;
+    if (sin_map_ != nullptr)
+    {
+      delete [] sin_map_;
+      delete [] cos_map_;
+    }
```

Comment:
Do we need a lock on this configure method? This gets called by `ScanShadowsFilter::reconfigureCB()` which (if i remember correctly) gets called directly in the roscpp worker thread handling an incoming reconfigure request. If someone uses this filter in a node that uses `ros::AsyncSpinner()`, then we might get incoming scans to process after we've deleted the sin and cos maps, but before we regenerate them. What do you think of adding a mutex member variable and grabbing it with a `std::lock_guard` here and when the maps are used?

Reply:
The maps are used in the isShadow method, which gets called (num_samples * window_size * 2) times per update, so adding a lock here is quite expensive. I would suggest letting the filter via its lock ensure that update and reconfigureCB never run concurrently. Do you agree?

Reply:
Yes, that makes sense.

Reply:
With the look-up tables moving from the detector to the filter (see your comment below), there's no need anymore for synchronization at the detector. But we added a mutex to the reconfigureCB method